pkg/controller/worker/machines_test.go

```diff
@@ -761,7 +684,8 @@ var _ = Describe("Machines", func() {
 
 					result, err := workerDelegate.GenerateMachineDeployments(ctx)
 					Expect(err).NotTo(HaveOccurred())
-					Expect(result).To(Equal(machineDeployments))
+					_ = machineDeployments
+					Expect(result).To(matchers.DeepEqual(machineDeployments))
```

Comment:
For consistency reasons you could use `matchers.DeepEqual` for the assertion in https://github.com/gardener/gardener-extension-provider-aws/blob/c9a30abe43a0523a9b25ab85935e141cc7ba6cf4/pkg/controller/worker/machines_test.go#L639